pkg/providers/gcp/provider_redis.go

```diff
@@ -511,3 +524,31 @@ func (p *RedisProvider) buildUpgradeInstanceRequest(instanceConfig *redispb.Inst
 	}
 	return nil
 }
+
+func (p *RedisProvider) checkRescheduleDeferMaintenance(ctx context.Context, redisClient gcpiface.RedisAPI, instance *redispb.Instance) error {
+	// Redis maintenance can be deferred up to 7 days
+	// If current Time minus lastScheduledMaintananceTime < 7 days (< 168 h), do nothing, otherwise - reschedule
+	if instance == nil {
+		return errorUtil.Errorf("redis Instance is null, can't reschedule redis maintenance")
+	}
+	lastScheduledMaintananceTime := instance.MaintenanceSchedule.StartTime
+	timeNow := time.Now()
+	if timeNow.Sub(lastScheduledMaintananceTime.AsTime()) < 166 {
```

Comment:
Hitting a nil pointer exception here on normal reconcile:

```
goroutine 2535 [running]:
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile.func1()
        /home/adam/go/src/github.com/integr8ly/cloud-resource-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:118 +0x1f4
panic({0x2502960, 0x46b0a00})
        /usr/lib/go/src/runtime/panic.go:884 +0x213
github.com/integr8ly/cloud-resource-operator/pkg/providers/gcp.(*RedisProvider).checkRescheduleDeferMaintenance(0x2f08b28?, {0x2f08b28, 0xc000ed1a70}, {0x2f09fa0, 0xc001cbb0a0}, 0xc0008f8ea0)
        /home/adam/go/src/github.com/integr8ly/cloud-resource-operator/pkg/providers/gcp/provider_redis.go:536 +0x7e
github.com/integr8ly/cloud-resource-operator/pkg/providers/gcp.(*RedisProvider).createRedisInstance(0xc0011a1040, {0x2f08b28, 0xc000ed1a70}, {0x2f0e690, 0xc0007ae690}, {0x2f09fa0?, 0xc001cbb0a0?}, 0xc0007aff80?, 0xc00106bd40)
        /home/adam/go/src/github.com/integr8ly/cloud-resource-operator/pkg/providers/gcp/provider_redis.go:188 +0xa32
github.com/integr8ly/cloud-resource-operator/pkg/providers/gcp.(*RedisProvider).CreateRedis(0xc0011a1040, {0x2f08b28, 0xc000ed1a70}, 0xc00106bd40)
        /home/adam/go/src/github.com/integr8ly/cloud-resource-operator/pkg/providers/gcp/provider_redis.go:102 +0x365
github.com/integr8ly/cloud-resource-operator/controllers/redis.(*RedisReconciler).Reconcile(0xc001289d00, {0x2f08b28, 0xc000ed1a70}, {{{0xc000f44048?, 0x0?}, {0xc00102c510?, 0x40fb47?}}})
        /home/adam/go/src/github.com/integr8ly/cloud-resource-operator/controllers/redis/redis_controller.go:175 +0x85e
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile(0x2f08a80?, {0x2f08b28?, 0xc000ed1a70?}, {{{0xc000f44048?, 0x28c9580?}, {0xc00102c510?, 0xc000e50d88?}}})
        /home/adam/go/src/github.com/integr8ly/cloud-resource-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:121 +0xc8
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc001008640, {0x2f08a80, 0xc000fc59a0}, {0x260c2c0?, 0xc001cba4c0?})
        /home/adam/go/src/github.com/integr8ly/cloud-resource-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:320 +0x309
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc001008640, {0x2f08a80, 0xc000fc59a0})
        /home/adam/go/src/github.com/integr8ly/cloud-resource-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:273 +0x1d9
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2()
        /home/adam/go/src/github.com/integr8ly/cloud-resource-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:234 +0x85
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2
        /home/adam/go/src/github.com/integr8ly/cloud-resource-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:230 +0x587
exit status 2
```

Reply:
Thank you very much Adam, I will check